python/tk_multi_workfiles/step_list_filter.py

```diff
@@ -99,18 +115,74 @@ def _cache_step_list(cls):
         Retrieve all Steps from Shotgun and cache them, if they were not already
         cached. Do nothing if they were already cached.
         """
-        if cls._step_list is None:
-            shotgun = sgtk.platform.current_bundle().shotgun
-            sg_steps = shotgun.find(
-                "Step",
-                [],
-                ["code", "entity_type", "color"],
-                order=[{"field_name": "code", "direction": "asc"}],
+        if cls._step_list is not None:
+            return
+
+        bundle = sgtk.platform.current_bundle()
+        shotgun_api = bundle.shotgun
+        # Use the engine's Shotgun instance for schema calls
+        engine_shotgun_api = sgtk.platform.current_engine().shotgun
+        limit_steps_to_project_visible = bundle.get_setting(
+            "project_visible_pipeline_steps", False
+        )
+        current_project = getattr(bundle.context, "project", None)
```

Comment:
in this case, the context will always have a project so you don't need to store it in a variable, you can just call bundle.context.project where you need to access the context project